### PR TITLE
LGA-2287 - Create Unit Tests for Assign Suggest View

### PR DESCRIPTION
**Pull request by jogilder**
## What does this pull request do?

There were no tests for the assign suggest view. This view failed when django rest framework was updated, decision made to add in extra coverage.

This is the view associated with clicking "assign" when creating and assigning a case on the HGS. You can click the tab and it hits the api even if there are validation errors from the frontend. Hence needing to check no category options.

It returns a response with the format:
`{"suggested_provider": *, "as_of": "date_time",
"suitable_providers":
 [{"name": "*", "id": pk, "short_code": "*",
"telephone_frontdoor": "*", "telephone_backdoor": "0845 122 8677"}, ...]}`

If the case has a category assigned then suitable_providers will be for only that category. 
If there is no category assigned to a case then suggested_provider will be None and suitable_providers will list all providers.
If the site is hit out of hours then once again suggested_provider will be None, suitable_providers will depend on whether a category has been selected.

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [ ] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: cla_backend/apps/call_centre/tests/api/test_case_api.py [hunk 1]
```diff
@@ -285,6 +286,106 @@ def test_patch_provider_notes_not_allowed(self):
...
+        else:
+            name = response.data["suggested_provider"]["name"]
+            self.assertIn(name, [unicode(x.name) for x in self.suggested_cat_providers])
+        #  now check the suitable_providers
+        if category_assigned:
+            suitable_providers = [ProviderSerializer(p).data for p in self.suggested_cat_providers]
+            self.assertEqual(response.data["suitable_providers"], suitable_providers)
+            # this should be all of the providers in suggested category
+
+            # self.assertItemsEqual(response.data["suitable_provider"], law_category=self.providers["category_id"]=])
```
Comment (psweeting1) [inline cla_backend/apps/call_centre/tests/api/test_case_api.py:380]: Do you need this commented out line of code?

File: cla_backend/apps/call_centre/tests/api/test_case_api.py [hunk 2]
```diff
@@ -285,6 +286,106 @@ def test_patch_provider_notes_not_allowed(self):
...
+        make_recipe(
+            "cla_provider.provider_allocation", weighted_distribution=0.5, provider=cat2_provider1, category=category2
+        )
+        cat2_provider2 = make_recipe("cla_provider.provider", active=True)
+        make_recipe(
+            "cla_provider.provider_allocation", weighted_distribution=0.5, provider=cat2_provider2, category=category2
+        )
+
+        helper1 = ProviderAllocationHelper()
+        self.suggested_cat_providers = helper1.get_qualifying_providers(category1)
```
Comment (psweeting1) [inline cla_backend/apps/call_centre/tests/api/test_case_api.py:323]: My IDE flags issues with some of the Instance attributes in suggested_cat_providers defined.

These instances need fields within the class

```
def __init__(self):
   self.suggested_cat_providers = None
   self.suggested_category = None
   self.cat_eligibility_check = None
```

**Review by psweeting1 — APPROVED** [review]
LGTM. Just address those tiny reviews and then you're all good to go.

File: cla_backend/apps/call_centre/tests/api/test_case_api.py (hunk 1 above)
Comment (jogilder) [reply cla_backend/apps/call_centre/tests/api/test_case_api.py:380]: removed

**Pull request merged**